File: W2/Implement Stack using Queues.dart
class MyStack {

  List<int> s=[];
  MyStack() {

  }
  
  void push(int x) {
    s.add(x);
  }
  
  int pop() {
    int last = s.last;
    s.removeLast();
    return last;
  }
  
  int top() {
    return s.last;
  }
  
  bool empty() {
    return s.length ==0;
  }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = MyStack();
 * obj.push(x);
 * int param2 = obj.pop();
 * int param3 = obj.top();
 * bool param4 = obj.empty();
 */

File: W2/Implement Stack using Queues.cpp
class MyStack {
public:
    vector<int>s;
    MyStack() {
        
    }
    
    void push(int x) {
      s.push_back(x);
    }
    
    int pop() {
        int last=s[s.size()-1];
        s.pop_back();
        return last;
    }
    
    int top() {
        return s[s.size()-1];
    }
    
    bool empty() {
        return s.empty();
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */